Compression Rename ht256 to ht

TodicaIonut opened the pull request:
Rename ht256 to ht 256-scanline chunks.

<!-- readthedocs-preview openexr start -->
Website preview: https://openexr--1975.org.readthedocs.build/en/1975/
<!-- readthedocs-preview openexr end -->

kdt3rd committed:
Fix usage of utf-8 filenames for windows

the std::filesystem::path mechanism is reasonable, but needs to be told
it to look for a utf-8 string, and is not automatically that, and the
standards committee must have realized there were issues and change the
api in c++20

Signed-off-by: Kimball Thurston <kdt3rd@gmail.com>

kdt3rd committed:
update all upload-artifact usage to v4 (#1970)

* update all upload-artifact usage to v4

Signed-off-by: Kimball Thurston <kdt3rd@gmail.com>

* replicate unique name change

Signed-off-by: Kimball Thurston <kdt3rd@gmail.com>

* Update .github/workflows/python-wheels-publish-test.yml

-${{ env.INSTALL_MANIFEST }} should definitely not get appended to the name, that was appropriate only for the install manifests in ci_steps.yml.

Signed-off-by: Cary Phillips <cary@ilm.com>

* Update .github/workflows/python-wheels-publish.yml

-${{ env.INSTALL_MANIFEST }} should definitely not get appended to the name, that was appropriate only for the install manifests in ci_steps.yml.

Signed-off-by: Cary Phillips <cary@ilm.com>

* Update .github/workflows/ossfuzz_workflow.yml

Signed-off-by: Cary Phillips <cary@ilm.com>

* Update .github/workflows/python-wheels.yml

Signed-off-by: Cary Phillips <cary@ilm.com>

* Update .github/workflows/scorecard.yml

Signed-off-by: Cary Phillips <cary@ilm.com>

* Update .github/workflows/scorecard.yml

Signed-off-by: Cary Phillips <cary@ilm.com>

---------

Signed-off-by: Kimball Thurston <kdt3rd@gmail.com>
Signed-off-by: Cary Phillips <cary@ilm.com>
Co-authored-by: Cary Phillips <cary@ilm.com>

peterhillman committed:
clean up exrmetrics help, simplify JSON output (#1973)

* clean up exrmetrics help, simplify JSON output

Signed-off-by: Peter Hillman <peterh@wetafx.co.nz>

* revert exrmetrics to returncode 0 with --help

Signed-off-by: Peter Hillman <peterh@wetafx.co.nz>

---------

Signed-off-by: Peter Hillman <peterh@wetafx.co.nz>

cary-ilm committed:
Add website page for exrmetrics (#1971)

* Add website page for exrmetrics

Signed-off-by: Cary Phillips <cary@ilm.com>

* add missing backticks

Signed-off-by: Cary Phillips <cary@ilm.com>

* usage examples

Signed-off-by: Cary Phillips <cary@ilm.com>

* run description

Signed-off-by: Cary Phillips <cary@ilm.com>

* Use a pool...

Signed-off-by: Cary Phillips <cary@ilm.com>

---------

Signed-off-by: Cary Phillips <cary@ilm.com>

TodicaIonut committed:
compression ht256 to ht

Signed-off-by: Todica Ionut <todicaionut2000111@gmail.com>

TodicaIonut committed:
compression ht256 to ht

Signed-off-by: Todica Ionut <todicaionut2000111@gmail.com>

TodicaIonut committed:
compression ht256 to ht

Signed-off-by: Todica Ionut <todicaionut2000111@gmail.com>

TodicaIonut committed:
compression ht256 to ht

Signed-off-by: Todica Ionut <todicaionut2000111@gmail.com>

TodicaIonut committed:
compression ht256 to ht

Signed-off-by: Todica Ionut <todicaionut2000111@gmail.com>

TodicaIonut committed:
compression ht256 to ht

Signed-off-by: Todica Ionut <todicaionut2000111@gmail.com>

TodicaIonut committed:
compression ht256 to ht

Signed-off-by: Todica Ionut <todicaionut2000111@gmail.com>

TodicaIonut committed:
compression ht256 to ht

Signed-off-by: Todica Ionut <todicaionut2000111@gmail.com>

TodicaIonut committed:
compression ht256 to ht

Signed-off-by: Todica Ionut <todicaionut2000111@gmail.com>

TodicaIonut committed:
compression ht256 to ht

Signed-off-by: Todica Ionut <todicaionut2000111@gmail.com>

TodicaIonut committed:
compression ht256 to ht

Signed-off-by: Todica Ionut <todicaionut2000111@gmail.com>

TodicaIonut committed:
compression ht256 to ht

Signed-off-by: Todica Ionut <todicaionut2000111@gmail.com>

TodicaIonut committed:
compression ht256 to ht

Signed-off-by: Todica Ionut <todicaionut2000111@gmail.com>

TodicaIonut committed:
compression ht256 to ht

Signed-off-by: Todica Ionut <todicaionut2000111@gmail.com>

TodicaIonut committed:
Fix Typo

Signed-off-by: Todica Ionut <todicaionut2000111@gmail.com>

cary-ilm committed:
Add HTJ2K news item (#1976)

Signed-off-by: Cary Phillips <cary@ilm.com>

kdt3rd committed:
fix potential buffer overwrite with zip data (#1974)

If the zipped data unpacks to a buffer which is too large, but still
within the scratch buffer size, could potentially write past the end of
the buffer

Signed-off-by: Kimball Thurston <kdt3rd@gmail.com>

kdt3rd committed:
Fix bad pointer math

In SSE, we were incrementing a pointer as if it were a uint16_t, not an
si128, causing out of bounds accesses on non-block aligned chunks.

Similarly, simplify the alignment logic and check return values and
pointers in related accesses (non-block aligned linear lut lookup).

Signed-off-by: Kimball Thurston <kdt3rd@gmail.com>

kdt3rd committed:
fix crash if user does not provide memory when filling (#1980)

for deep frame buffers, when filling, if the caller does not provide
output memory for a particular sample set, do not crash.

Signed-off-by: Kimball Thurston <kdt3rd@gmail.com>

kdt3rd committed:
avoid large allocations (#1979)

If the chunk table is unable to provide the specified data window, do
not allocate the frame buffer (or cache framebuffer) to that size

Signed-off-by: Kimball Thurston <kdt3rd@gmail.com>

palemieux commented:
@TodicaIonut What is your rationale for shortening `HT256` to `HT`? `256` felt relevant to me since the number of scanlines is hard-coded in the definition.

cary-ilm commented:
It doesn't seem essential to include the scanline count in the name, but "ht" does seem a bit brief. Other codecs include clarifying numbers that aren't scanlines (`pxr24`, `b44`). 

I'd like to propose `htj2k` as a more descriptive term. If anything, the "j2k" seems more important than the "ht". 

That can assume a default scanline count of 256, and should we ever add another type with a different number of scanlines, we can include the scanline count in it, e.g. `htj2k32`.

kdt3rd committed:
Address issues with small exr files and header parse

If we know the file size, we should truncate the header parse to only
read the file at most. Otherwise, if the file size is unknown (stream)
then be conservative (and slower) and read 1 byte at a time for the
header parse. Once we know the file, that should be free to read chunks

Signed-off-by: Kimball Thurston <kdt3rd@gmail.com>

kdt3rd committed:
Adjust the clamping on the dwa compression (#1986)

* Adjust the clamping on the dwa compression

The dwa compression is not truly a quality level like other things, so a
maximal value is controlling quantization, not quality. A value of 0 is
allowed, but negative is not.

Signed-off-by: Kimball Thurston <kdt3rd@gmail.com>

* improve tests for dwa compression level set

Signed-off-by: Kimball Thurston <kdt3rd@gmail.com>

---------

Signed-off-by: Kimball Thurston <kdt3rd@gmail.com>

mattyjams committed:
exrcheck: update CMakeLists.txt to install the tool (#1983)

* add an OPENEXR_INSTALL_DEVELOPER_TOOLS option

This new option will control whether or not the tools
considered to be "developer" tools should be installed.
These are tools useful for developing and debugging OpenEXR
itself that might not be suitable for distribution to end
users. exrcheck is currently the only tool considered to be
a developer tool.

Signed-off-by: Matt Johnson <matt.johnson@epicgames.com>

* exrcheck: update CMakeLists.txt to enable developer tool installation

This makes the CMakeLists.txt for exrcheck look like other
tools like exrinfo and uncomments the CMake call to install
the tool. One difference though is that exrcheck is considered
a developer tool, and as such is only installed when the
OPENEXR_INSTALL_DEVELOPER_TOOLS option is enabled.

Signed-off-by: Matt Johnson <matt.johnson@epicgames.com>

---------

Signed-off-by: Matt Johnson <matt.johnson@epicgames.com>

TodicaIonut committed:
Merge branch 'AcademySoftwareFoundation:main' into patch-2

TodicaIonut commented:
Close, Compression changing HT256 to HT? @palemieux 

> What is your rationale for shortening HT256 to HT? 256 felt relevant to me since the number of scanlines is hard-coded in the definition.

cary-ilm commented:
@TodicaIonut, our discussion in the project steering committee meeting led to the decision to use the name `htj2k`.

Our plan is to make a 3.3.3 patch release in the next week or so, then merge the `htj2k-beta` branch into `main`, where it will sit until the next non-patch release in August/September, which will be either 3.4 or possibly 4.0, we haven't decided that yet. That release will be the first one to include the htj2k support. But it will be available in the main branch within a week or so.
